### DISCUSSION TITLE
Would it be possible to have an option to set the maximum VRAM usage instead of trying to guess number of layers?

### DISCUSSION THREAD

**Discussion by TiagoTiago** [KoboldAI/KoboldAI-Client#257] [Ideas]
The trial and error process is a bit annoying...

**Comment by henk717** [comment]
We don't know the vram in advance since it depends on the model and further settings.

**Reply by TiagoTiago** [reply]
What if you start with just a small number of layers, and then extrapolate from that? Or do layers have wildly different sizes inside a model? Do you need to know the number of layers before you start sending them to the GPU, or could you check how much VRAM is used after each layer is added, and stop when the next layer would cross the threshold, switching to the CPU from that point onwards?
